packages/shared/components/popups/GovernanceCastVote.svelte

```diff
@@ -41,11 +45,22 @@
             votingAction = VotingAction.Cast
         } else if (currentVoteValue !== nextVote.value) {
             votingAction = VotingAction.Change
-        } else {
+        } else if (hasReceivedFundsSinceLastVote()) {
             votingAction = VotingAction.Merge
+        } else {
+            votingAction = VotingAction.Stop
         }
     }
 
+    const hasReceivedFundsSinceLastVote = (): boolean => {
```

Comment:
Maybe we can try making this function as crash proof as possible? Something like (just an example)
```ts
const hasReceivedFundsSinceLastVote = (): boolean => {
    const { trackedParticipations } = $participationOverview?.find(
        ({ accountIndex }) => accountIndex === $selectedAccount?.index
    ) ?? {}
    const { amount } = trackedParticipations?.[eventId]?.slice(-1)?.[0] ?? {}
    return amount !== $selectedAccount?.rawIotaBalance
}
```

Reply:
For sure! I need to work on my ability to make stuff crash proof. I'm not used to using the `?.` operator, so thanks for pointing this out!